plugins/poll/lib/irv.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+class DiscoursePoll::Irv
+  def self.irv_outcome(poll_id)
+    poll = Poll.find(poll_id)
+    options = PollOption.where(poll_id: poll_id).map { |hash| { id: hash.digest, html: hash.html } }
+
+    ballot = []
+    PollVote
+      .where(poll_id: poll_id)
+      .pluck(:user_id)
+      .uniq
+      .each do |user_id|
+        ballot_paper = []
+        PollVote
+          .where(poll_id: poll_id, user_id: user_id)
+          .order(rank: :asc)
+          .each { |vote| ballot_paper << vote.poll_option.digest if vote.rank > 0 }
+        ballot << ballot_paper
+      end
+    DiscoursePoll::Irv.run_irv(ballot, options) if ballot.length > 0
+  end
+
+  def self.run_irv(starting_votes, options)
```

Comment:
this method is too complex

Reply:
@jjaffeux could you please clarify what you mean here: too complex to complete within a reasonable time for the ajax request from the client, or too complex in general?  Obviously there is not much I can do about the basic complexity of the  "algorithm" of IRV, that's something I can't change.

Reply:
thanks @jjaffeux I will address those things and revert 👍

Reply:
I've addressed it here: https://github.com/discourse/discourse/pull/27155/commits/b844bf255b37f46f387932c348b33b53fdc4b794